client/interledger/js/components/asset_row.js

```diff
@@ -9,87 +12,177 @@ import AssetDetail from '../../../lib/js/react/components/asset_detail';
-        safeInvoke(handleAccountClick, activeAccount);
+        safeInvoke(handleAccountClick, account);
     },
 
-    handleEscrowClick(selectedAccount) {
+    handleActionClick(selectedAccount) {
         const {
-            asset,
-            activeAccount
+            account,
+            asset
         } = this.props;
-        
+
         const idToTransfer = {
             txid: asset.id,
             cid: 0
         };
 
-        const payloadToPost = {
-            'source': activeAccount,
-            'to': selectedAccount
-        };
+        if (asset.type === 'single-owner') {
+            AssetActions.escrowAsset({
+                idToTransfer,
+                payloadToPost: {
+                    source: account,
+                    to: selectedAccount
+                }
+            });
+        } else if (asset.type === 'multi-owner') {
+            AssetActions.fulfillEscrowAsset({
+                idToTransfer,
+                payloadToPost: {
+                    source: account,
+                    to: account
+                }
+            });
+        }
+    },
 
-        AssetActions.escrowAsset({
-            idToTransfer,
-            payloadToPost
-        });
+    getOperation() {
+        const {
+            account,
+            asset
+        } = this.props;
 
-        this.setState({ inEscrow: true });
+        let operation = 'transfer';
+        if (asset.hasOwnProperty('executeCondition') &&
+            account.vk === asset.executeCondition.public_key) {
+            operation = 'execute';
+        } else if (asset.hasOwnProperty('abortCondition') &&
+            account.vk === asset.abortCondition.public_key) {
+            operation = 'abort';
+        }
+        return operation;
     },
 
     render() {
         const {
+            account,
             accountList,
-            activeAccount,
+            actionMap,
             asset,
             isActive
         } = this.props;
 
         const {
-            inEscrow
+            expiresIn
         } = this.state;
 
+        const inBacklog = asset.hasOwnProperty('assignee');
+        const operation = this.getOperation();
 
         let actionsPanel = null;
-        if (isActive && activeAccount && accountList && !inEscrow) {
+        if (isActive && accountList && !inBacklog) {
+            const actionType = actionMap[`${asset.type}-${operation}`];
             actionsPanel = (
                 <AssetActionPanel
                     accountList={accountList}
-                    actionName="ESCROW"
-                    activeAccount={activeAccount}
-                    handleActionClick={this.handleEscrowClick} />
+                    actionMessage={actionType.actionMessage}
+                    actionName={actionType.actionName}
+                    activeAccount={account}
+                    handleActionClick={this.handleActionClick}
+                    selectAccounts={actionType.selectAccounts} />
```

Comment:
FYI, you could use `{...actionType}` here to avoid having to repeat `actionMessage`, `actionName`, and `selectAccounts`, but not a big deal here and it might even obscure what's getting passed.